### PR TITLE
DAOS-14440 build: Build on EL 8.6…

### PR DESCRIPTION
**Pull request by brianjmurrell**
But test on EL 8.8.

Build and test on Leap 15.5.
- update packaging support for Leap 15.5.

Update MOFED version so that newer OS versions are supported.

During provisioning move OS upgrade to after PR-repos are added so that
PR-repos can upgrade packages.

Use new repo-files mock repo names during RPM build.

Use fedora:latest for mock building.

Allow PR repos to replace packages from modules.

Move functions to the functions file.

Update RPM Test stage.
- Add "system-ram-reserved 4" to engine configuration.
- Auto-size SCM.
- Remove --recreate-superblocks from daos start.
- Collect logs and update status at the end of the stage.
- Update default provider.
- Define MODULEPATH for SUSE.
- Remove creation and moutning of tmpfs on /mnt/daos.
- Format the storage if needed.
- Add a 300s timeout to the server start and format.
- Display server stdout if startup fails.
- Use export POOL_NVME_SIZE=0.
- Archive DAOS logs.

Add a Jenkinsfile syntax checking githook.

Fix isort linter to ignore RPM _topdirs.

Update packaging/.

### Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by JohnMalmberg — APPROVED** [review]

**Review by ryon-jensen — APPROVED** [review]

File: utils/githooks/pre-commit.d/20-Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    exit 0
+fi
+
+echo "  Checking Jenkinsfile"
+
+HOST="${HOST:-build.hpdd.intel.com}"
+CURL_VERBOSE=${CURL_VERBOSE:-""}
+CURL_PROXY="${CURL_PROXY:+-x }${CURL_PROXY:-}"
+CURL_OPTS="$CURL_PROXY $CURL_VERBOSE -s"
+if ! output=$(curl $CURL_OPTS -s -X POST -F "jenkinsfile=<${1:-Jenkinsfile}" https://$HOST//pipeline-model-converter/validate); then
```
[the remark this answers is not in the document]
Comment (daltonbohning) [reply utils/githooks/pre-commit.d/20-Jenkinsfile:23]: Need to resolve this?

File: utils/githooks/pre-commit.d/20-isort.sh [hunk 2]
```diff
@@ -1,17 +1,25 @@
...
 
 set -ue
 
-if ! command -v isort > /dev/null 2>&1
-then
+if ! command -v isort > /dev/null 2>&1; then
     echo "No isort checking, install isort command to improve pre-commit checks"
+    . /etc/os-release
+    if [ "$ID" = "fedora" ]; then
+        echo "dnf install python3-isort"
```
Comment (daltonbohning) [inline utils/githooks/pre-commit.d/20-isort.sh:11]: Just FYI we should keep an eye on this. Installing `python3-isort` might not give the right isort version. But the version from requirements.txt might not be available either so the process is already not solid.

File: utils/githooks/pre-commit.d/20-isort.sh [hunk 3]
```diff
@@ -1,17 +1,25 @@
...
 
 echo "isort:"
 
 echo "  Running isort for python imports."
-isort --jobs 8 --check-only .
+isort_args=(--jobs 8 .)
+if ! isort --check-only "${isort_args[@]}"; then
+    echo "  isort check failed, run 'isort ${isort_args[*]}' to fix."
+    exit 1
+fi
```
Comment (daltonbohning) [inline utils/githooks/pre-commit.d/20-isort.sh:25]: Thank you :)

File: src/tests/ftest/daos_test/suite.single_host.yaml [hunk 4]
```diff
@@ -0,0 +1,159 @@
+# change host names to your reserved nodes, the
```
Comment (daltonbohning) [inline src/tests/ftest/daos_test/suite.single_host.yaml:1]: Where is `src/tests/ftest/daos_test/suite.single_host.yaml` being used?

Comment (brianjmurrell) [reply src/tests/ftest/daos_test/suite.single_host.yaml:1]: It was (is, could be in the future) used in the Test RPMs stages that right now that duplicate some of the cluster setup and test that is done by Ftest.  There are still some issues to be worked out with that but I thought it was worth leaving the YAML file there for future work in that area.

File: utils/githooks/pre-commit.d/20-Jenkinsfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (brianjmurrell) [reply utils/githooks/pre-commit.d/20-Jenkinsfile:23]: > Need to resolve this?

Maybe/probably.  Ideally in a future PR if that's agreeable as this one is apparently coming down to the wire.

**Review by daltonbohning — APPROVED** [review]

File: src/tests/ftest/daos_test/suite.single_host.yaml (hunk 4 above)
Comment (brianjmurrell) [reply src/tests/ftest/daos_test/suite.single_host.yaml:1]: The ultimate problem is that running ftest requires a minimum of two nodes and the Test RPMs test only allocates one.  We don't have VM clusters of less than 9 nodes but more than 1.

Comment (daltonbohning) [reply src/tests/ftest/daos_test/suite.single_host.yaml:1]: I think we could utilize `launch.py -e <path>` here. When you get around to testing with this config again, I could help you figure out the minimal config you need, to be applied on top of the "default" config.
Basically, if the default config is
```
foo: 1
bar: 2
```
And you use `launch.py -e <path>` where `path` is a config like this
```
bar: 3
```
The final test config used will be
```
foo: 1
bar: 3
```

**Pull request merged**